yeat/Snakefile

```diff
@@ -27,9 +27,25 @@ rule copyinput:
     input:
         read1=config["read1"],
         read2=config["read2"]
+    params:
+        file1="seq/input/{sample}_R1.fq",
+        file2="seq/input/{sample}_R2.fq"
     run:
-        copyfile(input.read1, output.read1)
-        copyfile(input.read2, output.read2)
+        if input.read1[-3:] == ".gz" and input.read2[-3:] == ".gz":
+            copyfile(input.read1, output.read1)
+            copyfile(input.read2, output.read2)
+        elif input.read1[-3:] != ".gz" and input.read2[-3:] == ".gz":
+            copyfile(input.read1, params.file1)
+            copyfile(input.read2, output.read2)
+            shell("gzip {params.file1}")
+        elif input.read1[-3:] == ".gz" and input.read2[-3:] != ".gz":
+            copyfile(input.read1, output.read1)
+            copyfile(input.read2, params.file2)
+            shell("gzip {params.file2}")
+        else:
+            copyfile(input.read1, params.file1)
+            copyfile(input.read2, params.file2)
+            shell("gzip seq/input/*")
```

Comment:
The if statement is broken up into four different conditions. Couldn't think of an easier and cleaner way to satisfy the snakemake rule.

Reply:
Also, in this same area, the only way I am checking the compression for each input file is their extension. I could take a look into the reading the first few [bytes](https://stackoverflow.com/questions/6059302/how-to-check-if-a-file-is-gzip-compressed) of the file. Overkill?

Reply:
I think trusting the extension is reasonable. This looks good to me.

Reply:
Updated code with Daniel's [suggestion](https://github.com/bioforensics/yeat/pull/31#discussion_r1015825043).